File: app/src/main/java/com/mmi/sdk/demo/java/activity/GradientPolylineActivity.java
package com.mmi.sdk.demo.java.activity;

import android.support.v7.app.AppCompatActivity;
import android.os.Bundle;

import com.mapbox.mapboxsdk.camera.CameraUpdateFactory;
import com.mapbox.mapboxsdk.constants.Style;
import com.mapbox.mapboxsdk.geometry.LatLng;
import com.mapbox.mapboxsdk.geometry.LatLngBounds;
import com.mapbox.mapboxsdk.maps.MapView;
import com.mapbox.mapboxsdk.maps.MapboxMap;
import com.mapbox.mapboxsdk.maps.OnMapReadyCallback;
import com.mmi.sdk.demo.R;
import com.mmi.sdk.demo.java.plugin.GradientPolylinePlugin;

import java.util.ArrayList;
import java.util.List;

public class GradientPolylineActivity extends AppCompatActivity implements OnMapReadyCallback {

    private MapboxMap mMapboxMap;
    private MapView mapView;
    private List<LatLng> listOfLatLng = new ArrayList<>();

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_gradient_polyline);
        mapView = findViewById(R.id.mapBoxId);
        mapView.getMapAsync(this);

        listOfLatLng.add(new LatLng(28.705436, 77.100462));
        listOfLatLng.add(new LatLng(28.705191, 77.100784));
        listOfLatLng.add(new LatLng(28.704646, 77.101514));
        listOfLatLng.add(new LatLng(28.704194, 77.101171));
        listOfLatLng.add(new LatLng(28.704083, 77.101066));
        listOfLatLng.add(new LatLng(28.703900, 77.101318));

    }

    @Override
    public void onMapReady(MapboxMap mapboxMap) {

        mapboxMap.setPadding(20, 20, 20, 20);
        LatLngBounds latLngBounds = new LatLngBounds.Builder()
                .includes(listOfLatLng)
                .build();
        mapboxMap.animateCamera(CameraUpdateFactory.newLatLngBounds(latLngBounds, 10));

        GradientPolylinePlugin animatedPolylinePlugin = new GradientPolylinePlugin(mapboxMap, mapView);
        animatedPolylinePlugin.createPolyline(listOfLatLng);
    }

    @Override
    public void onMapError(int i, String s) {

    }

    @Override
    protected void onStart() {
        super.onStart();
        mapView.onStart();
    }

    @Override
    protected void onStop() {
        super.onStop();
        mapView.onStop();
    }

    @Override
    protected void onDestroy() {
        super.onDestroy();
        mapView.onDestroy();
    }

    @Override
    protected void onPause() {
        super.onPause();
        mapView.onPause();
    }

    @Override
    protected void onResume() {
        super.onResume();
        mapView.onResume();
    }

    @Override
    public void onLowMemory() {
        super.onLowMemory();
        mapView.onLowMemory();
    }

    @Override
    protected void onSaveInstanceState(Bundle outState) {
        super.onSaveInstanceState(outState);
        mapView.onSaveInstanceState(outState);
    }
}

File: app/src/main/java/com/mmi/sdk/demo/kotlin/activity/GradientPolylineActivity.kt
package com.mmi.sdk.demo.kotlin.activity

import android.os.Bundle
import android.support.v7.app.AppCompatActivity
import com.mapbox.mapboxsdk.camera.CameraUpdateFactory
import com.mapbox.mapboxsdk.geometry.LatLng
import com.mapbox.mapboxsdk.geometry.LatLngBounds

import com.mapbox.mapboxsdk.maps.MapboxMap
import com.mapbox.mapboxsdk.maps.OnMapReadyCallback
import com.mmi.sdk.demo.R
import com.mmi.sdk.demo.kotlin.plugin.GradientPolylinePlugin
import kotlinx.android.synthetic.main.activity_gradient_polyline.*
import java.util.ArrayList

/**
 * Created by Saksham on 20/9/19.
 */
class GradientPolylineActivity : AppCompatActivity(), OnMapReadyCallback {

    private val listOfLatLng = ArrayList<LatLng>()

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_gradient_polyline)

        mapBoxId!!.getMapAsync(this)

        listOfLatLng.add(LatLng(28.705436, 77.100462))
        listOfLatLng.add(LatLng(28.705191, 77.100784))
        listOfLatLng.add(LatLng(28.704646, 77.101514))
        listOfLatLng.add(LatLng(28.704194, 77.101171))
        listOfLatLng.add(LatLng(28.704083, 77.101066))
        listOfLatLng.add(LatLng(28.703900, 77.101318))

    }

    override fun onMapReady(mapboxMap: MapboxMap) {

        mapboxMap.setPadding(20, 20, 20, 20)
        val latLngBounds = LatLngBounds.Builder()
                .includes(listOfLatLng)
                .build()
        mapboxMap.animateCamera(CameraUpdateFactory.newLatLngBounds(latLngBounds, 10))

        val animatedPolylinePlugin = GradientPolylinePlugin(mapboxMap, mapBoxId!!)
        animatedPolylinePlugin.createPolyline(listOfLatLng)
    }

    override fun onMapError(i: Int, s: String) {

    }

    override fun onStart() {
        super.onStart()
        mapBoxId!!.onStart()
    }

    override fun onStop() {
        super.onStop()
        mapBoxId!!.onStop()
    }

    override fun onDestroy() {
        super.onDestroy()
        mapBoxId!!.onDestroy()
    }

    override fun onPause() {
        super.onPause()
        mapBoxId!!.onPause()
    }

    override fun onResume() {
        super.onResume()
        mapBoxId!!.onResume()
    }

    override fun onLowMemory() {
        super.onLowMemory()
        mapBoxId!!.onLowMemory()
    }

    override fun onSaveInstanceState(outState: Bundle) {
        super.onSaveInstanceState(outState)
        mapBoxId!!.onSaveInstanceState(outState)
    }
}
